lib/order.rb

```diff
@@ -0,0 +1,102 @@
+require 'csv'
+
+# Create a class called Order. Each new Order should include the following attributes:
+#   ID, a number (read-only)
+#   A collection of products and their cost
+#   An instance of Customer
+#   A fulfillment_status
+class Order
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+  
+  def initialize (id, products, customer, fulfillment_status = :pending)  # If no fulfillment_status is provided, it will default to :pending
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+    
+    # fulfillment_status, one of the following symbols :pending, :paid, :processing, :shipped, or :complete
+    valid_statuses = %i[pending paid processing shipped complete]
+    valid_statuses.include?(fulfillment_status)
+    
+    # If a status is given that is not one of the above, an ArgumentError should be raised
+    if !valid_statuses.include?(fulfillment_status)
+      raise ArgumentError.new "Invalid fulfillment status."
```

Comment:
It's helpful to say what was invalid when you raise an error.

```suggestion
      raise ArgumentError.new "Invalid fulfillment status: #{fulfillment_status}."
```